### ISSUE TITLE
about your data file

### ISSUE THREAD

**Issue by yanfeng9611** [JRC1995/3D-CNN-Keras#1]
Hello, I am only a student and currently studying this direction.Thank you very much for the open source of this code, let me open my mind a lot.But there were some problems when I tried to reproduce your project.I don't have the corresponding data, may I get your data file?Looking forward to your reply.

**Comment by JRC1995** [comment]
The data set may not be exactly open access. I was given access to it for an interview. I am not sure if I am allowed to share it. Furthermore, it is too big (8 GB) for me to upload anywhere. My upload speed is also slow. As a substitute, there seems a lot of medical datasets here: https://www.via.cornell.edu/databases/ (found from google). You may find some more from Google. 
Also, the code is pretty old. I am not sure if it will run in new keras. But hopefully, it wouldn't require too many changes.

**Comment by yanfeng9611** [comment]
Ok, thanks for your reply. Now I am trying to make the data set into h5py file by myself.But there are problems.After reading the official document of h5py, I am trying to do it myself. However, I can only add patch_id and data.I found that you have a line of code: data [patch_id] [' metainfo] [' label '] [...].==1. I don't know how to add the label, and I'm looking forward to your help.

**Comment by JRC1995** [comment]
Which data set are you using? You can add the labels if you have labels in the first place. I don't know if the data in the link comes with labels for supervised training or not. Also, you can also consider changing the data processing part of the code to better suit whatever data format you are using. 

BTW, If you simply want to try image classification or something like that, you can look for cifar10 and cifar100. Those are popular openly available labeled datasets I know of. They are not in the medical domain though. I am not very experienced with image-related datasets in the medical domain.

**Comment by yanfeng9611** [comment]
Thank you very much for your reply.

I downloaded the LUNA data set and tried to do classification training by myself. But there are always problems when loading data. Admittedly, keras's ImageDataGenerator is a better method, but elsewhere we've seen people say that loading files like this doesn't make much use of memory. Before entering the contents, enter the contents in the candidates.h5 file before network training. I thought this was a great way to format the LUNA dataset, so I decided to do it myself.

In the process of production, I can only add patch_id and corresponding data, but cannot add label. It is similar to: f.patch dataset (file_name[I],data=img). Print (data[patch_id][...]). This is unlabeled and cannot be used for training, so hope to get your help.

**Comment by JRC1995** [comment]
If the data is not labeled you probably have to try some different data. My data was about a simple image classification. Given a temporal series of CT scan it has to classify whether there were ever any nodule or not. From what I checked LUNA annotations seemed to have mostly segmentation related data where the "label" is the portion where the nodule is present. Candidates csv file in LUNA seems to contain candidate locations of nodules and its class. Its class can then be used as labels but note it pairs only candidate locations - a specific portion of an image in CT scan data with the label. But this program wants inputs to be the ct scan data itself not the specific x,y,z co-ordinate. One thing that may be possible is to find which CT scan data the coordinates refer to. Then find those CT scan image-series data of which ALL candidates nodules in candidates csv are marked 0 and assign a label 0 to them. And then mark any ct scan image series data of which any one or more candidate nodules have a class 1 as 1. Don't know how the quality of the final data will be - i.e if there will be sufficient balance of positive and negative examples. I don't know how exactly the files in LUNA are related and how candidate nodule coordinaes can be mapped to scan images (may be by taking scan inage name into account). I only glanced at LUNA descriptions some of my understanding about the data may be wrong.
